CONTRIBUTION.md

```diff
@@ -28,7 +29,353 @@ These commands can be executed from the project root:
+		return
+	}
+
+	r.client = apiClient
+	tflog.Info(ctx, "Foo bar client configured")
+}
+
+// Schema defines the schema for the resource.
+func (r *barResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	descriptions := map[string]string{
+		"main":               "Foo bar resource schema.",
+		"id":                 "Terraform's internal resource identifier. It is structured as \"`project_id`,`bar_id`\".",
+		"project_id":         "STACKIT Project ID to which the bar is associated.",
+		"bar_id":             "The bar ID.",
+		"my_field":           "My field description.",
+		"my_read_only_field": "My read-only field description.",
+	}
+
+	resp.Schema = schema.Schema{
+		Description: descriptions["main"],
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: descriptions["id"],
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Description: descriptions["project_id"],
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					// Validators can be used to validate the values set to a field
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"bar_id": schema.StringAttribute{
+				Description: descriptions["bar_id"],
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					// "RequiresReplace" makes the provider recreate the resource when the field is changed in the configuration
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"my_field": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					// "UseStateForUnknown" can be used to copy a prior state value into the planned value. It should be used when it is known that an unconfigured value will remain the same after a resource update
+					stringplanmodifier.UseStateForUnknown(),
+				}
+			},
+			"my_read_only_field": schema.StringAttribute{
+				Computed: true,
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *barResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) { // nolint:gocritic // function signature required by Terraform
+	var model Model
+	diags := req.Plan.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	projectId := model.ProjectId.ValueString()
+	barId := model.BarId.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "bar_id", barId)
```

Comment:
usually this is only set after getting the response, where the ID is returned

Reply:
Good catch, same as https://github.com/stackitcloud/terraform-provider-stackit/pull/322#discussion_r1557657006